rfd/00xx-kubernetes-access-cluster-discovery.md

```diff
@@ -0,0 +1,362 @@
+
+The necessary IAM permissions required for calling those two methods are:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "eks:DescribeCluster",
+                "eks:ListClusters"
+            ],
+            "Resource": "*" # can be limited
+        }
+    ]
+}         
+```
+
+Besides the previous IAM policies required for the discovery service, the discovery service must also have the `sts:GetCallerIdentity` IAM permission to generate an access token to Kubernetes API.
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "sts:GetCallerIdentity"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+With cluster details such as API endpoint and CA, Teleport creates an access token to gain access to the cluster. The access is given by a token generated by [AWS IAM authenticator][awsiamauthenticator] project. This project generates a short-lived user access token by mapping the user IAM username/role into [Kubernetes RBAC credentials][iamroleatch] ([eksctl IAM Mappings][iameksctl]).
+
+AWS IAM does not allow setting access permissions to any Kubernetes cluster. This means that it is not possible through the AWS API to guarantee access to one or several Kubernetes clusters. This is achieved by an extra project, [AWS IAM authenticator][awsiamauthenticator], that is installed, by default, in the EKS control plane which translates the IAM roles and users into Kubernetes roles or users. [AWS IAM authenticator][awsiamauthenticator] component receives a request when any user generates an access token for a cluster and, it searches in its database for any match between IAM user or role and the Kubernetes role. If the match is found, it returns a short-lived access token for the impersonated Kubernetes user or role, otherwise, it returns a token for `system:anonymous` group which has no access to resources.
+
+The [AWS IAM authenticator][awsiamauthenticator] database is a simple `configmap/aws-auth` stored in `kube-system` namespace. This configmap cannot be changed unless Teleport has write permissions to Kubernetes Configmaps, which can only happen if the Teleport role is already in the configmap.
+
+ The configmap has the following format:
+
+```yaml
+apiVersion: v1
+data:
+  mapRoles: |
+    - groups:
+      - {kube group}
+      rolearn: {IAM role}
+      username: {user name}
+  mapUsers: |
+    - groups:
+      - {kube group}
+      userARN: {IAM user}
+      username: {user name}
+...
+```
+
+For Teleport to be able to connect and operate the Kubernetes cluster, it is required that the `configmap/aws-auth` maps the Teleport role into a Kubernetes group.
```

Comment:
By "Teleport role" - do you mean the AWS IAM role that Teleport Kubernetes agent is running as?